### ISSUE TITLE
Cannot do migrations for the IdentityServer4's database contexts in my project.

### ISSUE THREAD

**Issue by ilap** [PomeloFoundation/Pomelo.EntityFrameworkCore.MySql#437]
### Steps to reproduce
There is more than one DB context in my project and two comes /w IdentityServer4: 

- IdentityServer4.EntityFramework.DbContexts.ConfigurationDbContext
- IdentityServer4.EntityFramework.DbContexts.PersistedGrantDbContext

Create s simple solution /w EF, Identity, MsSQL and IdentityServer4.

Command I ran:

`dotnet ef migrations add Initial --context ConfigurationDbContext`

Got the following:
``` bash
Your target project 'Identity.API' doesn't match your migrations assembly 'IdentityServer4.EntityFramework'. Either change your target project or change your migrations assembly.
Change your migrations assembly by using DbContextOptionsBuilder. E.g. options.UseSqlServer(connection, b => b.MigrationsAssembly("ProjectName")). By default, the migrations assembly is the assembly containing the DbContext.
Change your target project to the migrations project by using the Package Manager Console's Default project drop-down list, or by executing "dotnet ef" from the directory containing the migrations project.
```

### The issue
The Pomelo's MySql does not have the MSSql's MigrationsAssembly equivalent in DbContextOptionsBuilder, to be able to do migrations on those DB contexts to create the relevant tables in the database.

In my project for MsSQL I use the following code to change my migrations assembly by using the DbContextOptionsBuilder.
```
var migrationsAssembly = typeof(Startup).GetTypeInfo().Assembly.GetName().Name;
...
options.ConfigureDbContext = b => b.UseMySql(connString,
                         sqlServerOptionsAction: sqlOptions =>
                        {
                             sqlOptions.MigrationsAssembly(migrationsAssembly);
```
MySQL version: 5.7
Operating system: macOS High Sierra
Pomelo.EntityFrameworkCore.MySql version: 2.0.01

Other details about my project setup:
Simple Web API project /w IdentityServer4 and was trying to migrate from MsSql to MySql.

**Comment by yukozh** [comment]
@ilap I don't think migrate from mssql to mysql is a valid operation.

**Comment by ilap** [comment]
Sorry, I think I was not clear. I did mean migrating from MsSQL to MySQL (with the existing data). I just changed the backend, means empty new MySQL database, and then wanted to generate new tables using dotnet ef CLI for EF (which was successful and the tables were created) and for the IdentityServer4, which were not successful as the IdentityServers4 assemblies are named IdentityServer4.EntityFramework.DbContexts.ConfigurationDbContext and IdentityServer4.EntityFramework.DbContexts.PersistentGrantDbContext respectively (Means I did not created any DbContext for these in my Project, but wanted to use the original EF DbContexts existing in IS4).

**Comment by ilap** [comment]
Sorry typo above, I did not mean migrating from MSSQL to MySQL.

**Comment by yukozh** [comment]
So, could you double check the target project you selected.

**Comment by ilap** [comment]
The target project is the same. I have just unconfigured the MsSQL in the same project and configured the MySQL backend /w EF, and wanted to create the schemas for Identity and IS4. Identity created the tables fine, as I have an ApplicaitonDbContext in my project, however, the schemas, for IS4's Configuration and Operations dbContext, were not created, as they use the IS4's original assembly name (e.g. IdentityServer4.EntityFramework.DbContexts.ConfigurationDbContext). That's why there is a MigrationsAssembly in MSSQL's DbContextOptionsBuilder, which uses my project's assembly name for the migrations. But, this does not exist in MySQL. If I revert that changes every dotnet ef commands work /w MSSQL.

**Comment by yukozh** [comment]
Sorry, I could not understand what you say.

**Comment by ilap** [comment]
Hi,

Sorry it's fixed now, with the below code as the MigrationsAssembly is a member of the System.Data.Entity.Migrations's DbMigrationsConfiguration class.

The issue was that the IdentityServer4's nuget package, that I use in my project, contains 2 DbContext definitions, one for its ConfigurationalStore and one for the OperationolStore, that comes /w the IdentityServer4.EntityFramework.DbContexts.ConfigurationDbContext/PersistedGrantDbContext namespaces. Therefore, my project could not do any initial migration because of the Assembly info mismatch of the dbcontexts. The resolution is to change migrations' Assembly info, to my project's one, what the ` sqlOptions.MigrationsAssembly(migrationsAssembly); ` does.

But, as I can see that the EnableRetryOnFailure has not implemented in the v2.0.0.1, but I can see it in the latest commit of the MySqlDbContextOptionsBuilder class.

```
                {
                    options.ConfigureDbContext = builder => builder.UseMySql(connectionString,
                                     sqlOptions =>
                                     {
                                         sqlOptions.MigrationsAssembly(migrationsAssembly);
                                         //sqlOptions.EnableRetryOnFailure(maxRetryCount: 10, maxRetryDelay: TimeSpan.FromSeconds(30), errorNumbersToAdd: null);
                                     });
                })
```

**Comment by JoseMOrtizR** [comment]
Thank you ilap. Very usefull.
